### PR TITLE
Added Coveralls.io report generation and badge

### PR DESCRIPTION
**Pull request by Lewiscowles1986**
Addresses #54 

Work that needs to be done by an owner of imanee/imanee 
Sign up to / add repository to coveralls.io (Markdown for badge already added)

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -23,12 +23,4 @@
             </exclude>
         </whitelist>
     </filter>
-    <logging>
-        <log type="coverage-html" target="./build/coverage" title="Imanee"
-             charset="UTF-8" yui="true" highlight="true" lowUpperBound="5"
-             highLowerBound="70" />
-        <log type="coverage-clover" target="./build/logs/clover.xml" />
-        <log type="junit" target="./build/logs/junit.xml"
```
Comment (mikeSimonson) [inline phpunit.xml]: I would just remove the junit target as it's never going to be used.

**Comment by Lewiscowles1986** [comment]
:+1:  thx, actually removed the whole of the <logging> section from phpunit as the .travis.yml handles coverage after running tests :wink:

**Comment by mikeSimonson** [comment]
Well I think it's a better idea to do it the other way around.
So that when you run the tests on your machine you can see the coverage with the html output and you are not tied to much to travis.
Otherwise the travis configuration is going to be unreadable very fast.

**Comment by Lewiscowles1986** [comment]
@mikeSimonson noted, but the `.travis.yml` file is not very complex at all, and there is nothing to stop an individual developer adding one entry to logging to output html coverage to `phpunit.xml`, if that fits their use-case... This is more about showing to people who do not have the code on their local machine

 * that you are testing your code
 * how valuable your tests are

With the intention of getting more people involved in imanee.

**Comment by mikeSimonson** [comment]
It's easier to on-board people if they can see the difference in coverage on their machine directly.
Instead of having them individually configure phpunit, you can configure it for everyone and for travis at the same time.

If you want to involve more people it's the way to go.

**Comment by Lewiscowles1986** [comment]
@mikeSimonson just think about what you said there in the context of what I am saying...

> It's easier to on-board people if they can see the difference in coverage on their machine directly.

as part of a response to

> This is more about showing to people ***who do not have the code on their local machine***

**i.e.** they haven't downloaded the library, why should they? How can imanee pursuade more coders to use the lib (because let's face it, the more people in front of something like this, the more features, and refinement to make the idea awesome). 

Maybe another library displays coverage, so they feel it is more mature in terms of tested features and known code quality. In any case the second part is what this does, that you are arguing against...

> Instead of having them individually configure phpunit, you can configure it for everyone and for travis at the same time.

Please read the `.travis.yml` file which does `after_script` generate coverage to send to coveralls, so everyone can see it and you can machine hop for dev environments requiring that (so not BYOD etc).

Most people wont

**Comment by mikeSimonson** [comment]
>    This is more about showing to people who do not have the code on their local machine


By putting the configuration in the phpunit you ensure the code coverage is done in both case.

You can remove the option in the phpunit call in .travis and the code coverage is going to be run.

**Comment by erikaheidi** [comment]
So, I'm gonna merge this and if there are suggestions for changing it please open a new PR :)

**Comment by erikaheidi** [comment]
The coverage says "unknown", any ideas why?

**Comment by Lewiscowles1986** [comment]
It doesn't on your repo now :wink: takes a while to generate. Basically the % is important, but also is trying to ensure multiple tests are run per method to test as many use-cases as possible, and coveralls.io also shows this as x3 for example on the right of each line, which demonstrates that there should be n scenario's playing out per method.

It's really a cool tool, and can set targets for development, thanks for interest and time, and patience!

**Pull request merged**